pkg/certificate/secret.go

```diff
@@ -64,3 +65,50 @@ func (m *Manager) applyTLSSecret(service types.NamespacedName, keyPair *triple.K
 		return m.client.Update(context.TODO(), updateTLSSecret(secret, keyPair))
 	})
 }
+
+// checkTLS will verify that the caBundle and Secret are valid and can
+// be used to verify
+func (m *Manager) verifyTLSSecret(secretKey types.NamespacedName, caBundle []byte) error {
+	secret := corev1.Secret{}
+	err := m.get(secretKey, &secret)
+	if err != nil {
+		return errors.Wrapf(err, "failed getting TLS secret %s", secretKey)
+	}
+
+	keyPEM, found := secret.Data[corev1.TLSPrivateKeyKey]
+	if !found {
+		return errors.New("TLS key not found")
+	}
+
+	_, err = triple.ParsePrivateKeyPEM(keyPEM)
+	if err != nil {
+		return errors.Wrap(err, "failed parsing PEM TLS key")
+	}
+
+	certsPEM, found := secret.Data[corev1.TLSCertKey]
+	if !found {
+		return errors.New("TLS certs not found")
+	}
+
+	certs, err := triple.ParseCertsPEM(certsPEM)
+	if err != nil {
+		return errors.Wrap(err, "failed parsing PEM TLS certs")
+	}
+
+	cas := x509.NewCertPool()
```

Comment:
This part is already used in checkTLS(). can you put it in a function and it both?

Reply:
Yep, let me refactor both so we reuse some code.